core/src/test/scala/org/apache/spark/ExecutorAllocationManagerSuite.scala

```diff
@@ -272,17 +272,41 @@ class ExecutorAllocationManagerSuite
   test("ignore task end events from completed stages") {
```

Comment:
the name of this test is needs to be updated.  Maybe "properly handle task end events from completed stages"